### ISSUE TITLE
Maya crashes in AL's plugin profiling code when opening file

### ISSUE THREAD

**Issue by pilarmolinalopez** [Autodesk/maya-usd#127]
The following assert consistently fails when opening one of our scene files:
```
void Profiler::pushTime(const AL::usdmaya::ProfilerSectionTag* entry) {
      assert(MAX_TIMESTAMP_STACK_SIZE > m_stackPos);
```
This function is in AL's profiling code in the AL/usdmaya/CodeTimings.cpp file. MAX_TIMESTAMP_STACK_SIZE is equal to 16. The timestamp stack in AL's profiling code keeps overflowing.  Profiler::pushTime() is being called from ProxyShape::loadStage(). The scene causing the crash has multiple rigged references and proxyShapes. We have not been able to isolate the issue into a more simple scene yet.

This doesn't happen when we disable the UFE build.
**Disabling the UFE part of the build fixes the issue (-DCMAKE_WANT_UFE_BUILD=0)**

A few more notes about our environment:
- environment - recent dev branch or refactoring_sandbox
- AL plugin compiled with CMAKE_WANT_UFE_BUILD=1
- environment variable MAYA_WANT_UFE_SELECTION being set is irrelevant ( occurs with it enabled and also disabled )

**Comment by kxl-adsk** [comment]
Overflow can cause many issues in random places... @murphyeoin - should this stack size be increased? What makes the number of pushed start times go so high? (is it safe to run in parallel?)

**Comment by pilarmolinalopez** [comment]
Forgot to say, I increased the size up to 50 and it would still crash.
What I don't understand yet is why this only happens when UFE build is enabled. @kxl-adsk - Do you have any clues about that?

**Comment by kxl-adsk** [comment]
Does it mean that without UFE we are not overflowing this stack?

**Comment by pilarmolinalopez** [comment]
Correct. As I said in the description, if we compile with CMAKE_WANT_UFE_BUILD=0 the stack does not get overflowed. It overflows when we compile with CMAKE_WANT_UFE_BUILD=1. This happens regardless MAYA_WANT_UFE_SELECTION being set or not.

**Comment by kxl-adsk** [comment]
Thank you for clarification. We will need a way to reproduce it...and in the meantime, can you paste the full callstack when MAX_TIMESTAMP_STACK_SIZE assert is hit? (I expect to see some recursive calls in there)

**Comment by pilarmolinalopez** [comment]
Sure! I will do that when I'm back at work tomorrow. I will also keep trying to isolate the issue for you to be able to reproduce it.

**Comment by pilarmolinalopez** [comment]
hi @kxl-adsk  - This is the stack trace we are getting when that assert is hit:
[stackTrace.txt](https://github.com/Autodesk/maya-usd/files/3881068/stackTrace.txt)

**Comment by kxl-adsk** [comment]
Thank you @pilarmolinalopez - we clearly see the pattern of recursive calls caused by ProxyShapeHandler::dagPathToStage. I feel the data is not set clean before notification is sent from AL::usdmaya::nodes::ProxyShape::computeOutStageData.

**Comment by kxl-adsk** [comment]
Can you give it a try?

**Comment by pilarmolinalopez** [comment]
We have a fix for this that we are currently testing. We'll submit a PR soon if the fix is confirmed.
Thanks @kxl-adsk  for your help!

**Comment by cfmoore007** [comment]
Closing this out, as the code has been merged into dev.
